### ISSUE TITLE
Search Filters are lost after sorting or change page.

### ISSUE THREAD

**Issue by nikomiranda** [tabalinas/jsgrid#138]
I don't know if I'm doing something wrong but I have a server side ajax controller that sorts, page and filter based on request parameters. 

	  		$("#grid").jsGrid("search", filter).done(function() {
	  		   console.log("filtering completed");
	  		});

On this call, search is working correctly but If i sort using the column header or change page the filter is lost. (paging and sorting is working okey without filters.

**Comment by tabalinas** [comment]
I guess you have `pageLoading: true`. In this case changing a page or sorting requires reloading all data. Where filter data are coming from? Do you have a filter row? If not, then you can pass your filter to server right in the `controller.loadData` method. If you have filter row, then the values of filtering inputs should be set.

As explained in this issue https://github.com/tabalinas/jsgrid/issues/81
> For now you should restore the filter directly with jQuery (select elements by .jsgrid-filter-row and set values to inputs). Then you can call sort and filter with sorting will be applied.

**Comment by nikomiranda** [comment]
Yes, I have `pageLoading: true `. I don't use filter rows. I'm calling programmatic the method    `$("#grid").jsGrid("search", filter)` with a filter object I build (with extra parameters for the server-side controller), then inside the loaddata function I pass it to the server with ajax.
 
       ...
       pageLoading: true,
       controller: {
              loadData: function(filter) {
                   jQuery.extend(true, filter, savedSearchFilter);
                   var data=$.ajax({
                     url: "/listJson",
                    data: filter,
                    dataType: "json",
                    async: false,
                });
                return {
                    data: data.responseJSON.rows,
                    itemsCount: data.responseJSON.total
                 };   
         }
         ...

But when I click on the pager or a column header for sorting the loaddata method is called again but the filters I used on the search method are lost. So for now as a workaround I am saving in the *savedSearchFilter* object before calling search and then extending inside loaddata.

**Comment by tabalinas** [comment]
Yes, what you are doing I guess is the best workaround. Your scenario just was not considered - having filtering but turn off filter row.

**Comment by naskrina** [comment]
Looks like the method causing the trouble is this one
`sort: function() {
    return this._grid.loadData();
}`
It calls loadData() without parameters.
But it's still not fixed in the newest version.
